src/Microsoft.AspNetCore.Sockets.Client/IConnection.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.AspNetCore.Sockets.Client
 {
     public interface IConnection
     {
-        Task StartAsync(ITransport transport, HttpClient httpClient);
+        Task StartAsync(ITransportFactory transportFactory, HttpClient httpClient);
```

Comment:
Why did we make HttpClient part of the interface instead of a ctor parameter? The interface doesn't need to be tied to a particular http client implementation does it?